pkg/apis/config/v1alpha1/adminqos.go

```diff
@@ -88,4 +93,207 @@ type ReclaimedResourceConfig struct {
+	// +optional
+	MemoryHeadroomConfig *MemoryHeadroomConfig `json:"headroomPolicyConfig,omitempty"`
+
+	// CPUHeadroomConfig is a configuration for cpu headroom
+	// +optional
+	CPUHeadroomConfig *CPUHeadroomConfig `json:"cpuHeadroomConfig,omitempty"`
+}
+
+type MemoryHeadroomConfig struct {
+	// PolicyCanonicalConfig is a config for canonical memory headroom policy
+	// +optional
+	PolicyCanonicalConfig *MemoryHeadroomPolicyCanonicalConfig `json:"policyCanonicalConfig,omitempty"`
+}
+
+type MemoryHeadroomPolicyCanonicalConfig struct {
+	// MemoryUtilBasedConfig is a config for memory utilization based memory headroom policy
+	// +optional
+	UtilBasedConfig *MemoryUtilBasedConfig `json:"utilBasedConfig,omitempty"`
+}
+
+type MemoryUtilBasedConfig struct {
+	// Enabled is a flag to enable utilization based memory headroom policy
+	// +optional
+	Enabled *bool `json:"enabled,omitempty"`
+
+	// FreeBasedRatio is the estimation of free memory utilization, which can
+	// be used as system buffer to oversold memory.
+	// +optional
+	FreeBasedRatio *float64 `json:"freeBasedRatio,omitempty"`
+
+	// StaticBasedCapacity is the static oversold memory size by bytes
+	// +optional
+	StaticBasedCapacity *float64 `json:"staticBasedCapacity,omitempty"`
+
+	// CacheBasedRatio is the rate of cache oversold, 0 means disable cache oversold
+	// +optional
+	CacheBasedRatio *float64 `json:"cacheBasedRatio,omitempty"`
+}
+
+type CPUHeadroomConfig struct {
+	// PolicyUtilizationConfig is a config for utilization based cpu headroom policy
+	PolicyUtilizationConfig *CPUHeadroomPolicyUtilizationConfig `json:"policyUtilizationConfig,omitempty"`
+}
+
+type CPUHeadroomPolicyUtilizationConfig struct {
+	// TargetReclaimedCoreUtilization is the target reclaimed core utilization to be used for
+	// calculating the oversold cpu headroom
+	// +optional
+	TargetReclaimedCoreUtilization *float64 `json:"targetReclaimedCoreUtilization,omitempty"`
+
+	// MaxReclaimedCoreUtilization is the max reclaimed core utilization of reclaimed_cores pool,
+	// which is used to calculate the oversold cpu headroom, if zero means no limit
+	// +optional
+	MaxReclaimedCoreUtilization *float64 `json:"maxReclaimedCoreUtilization,omitempty"`
+
+	// MaxOversoldRate is the max oversold rate of cpu headroom to the actual size of
+	// reclaimed_cores pool
+	// +optional
+	MaxOversoldRate *float64 `json:"maxOversoldRate,omitempty"`
+
+	// MaxHeadroomCapacityRate is the max headroom capacity rate of cpu headroom to the total
+	// cpu capacity of node
+	// +optional
+	MaxHeadroomCapacityRate *float64 `json:"maxHeadroomCapacityRate,omitempty"`
+}
+
+type EvictionConfig struct {
+	// EvictionManagerConfig is the config for eviction manager
+	// +optional
+	EvictionManagerConfig *EvictionManagerConfig `json:"evictionManagerConfig,omitempty"`
+
+	// EvictionPluginsConfig is the config for all eviction plugins
+	// +optional
+	EvictionPluginsConfig *EvictionPluginsConfig `json:"evictionPluginsConfig,omitempty"`
```

Comment:
Do we need to expose implementation-related concepts such as manager and plugin to users?